### DISCUSSION TITLE
Speed with and without filer

### DISCUSSION THREAD

**Discussion by zheliazkov** [seaweedfs/seaweedfs#1835] [Q&A]
Hi guys, Hi @chrislusf 

I'm currently testing and benchmarking the seaweedfs and  I'm experiencing a big difference between speed tests done by the weed itself (weed benchmark) and other - parallel curl requests or FUSE mount. In a big difference I mean the internal benchmark is at least 10 times faster. 
The benchmark is writing/reading a lot (100000) of small files (100KB). I've tried multi-threaded execution also.

Have somebody tried this comparison and is somebody know what the bottleneck can be?

All the servers (master, volume, filer) are running on one machine and the client is another. Network is very fast, disk also, and CPUs on both of them are good.

I've tried to change the filer db to redis, but there is no improvement.

Thanks in advance.

BR,
Plamen

**Comment by chrislusf** [comment]
There are multiple techniques used in the benchmarking:
- Reuse http connections to the volume servers
- Cache the mapping of volume ids to volume server locations.
- Multi-threading

The small things will add up.

You can also use "weed filer.copy" to write the files and see the performance.
